test/app/on_shutdown.test.lua

```diff
@@ -0,0 +1,38 @@
+env = require('test_run')
+test_run = env.new()
+build_dir = os.getenv("BUILDDIR") .. "/test/app/"
+soext = (jit.os == "OSX" and "dylib" or "so")
+on_shutdownlib = "on_shutdownlib."..soext
+os.execute(string.format("cp %s/%s .", build_dir, on_shutdownlib))
+
+test_run:cmd('create server test with script="app/on_shutdown.lua"')
+test_run:cmd('start server test')
+test_run:cmd('switch test')
+
+module = require('on_shutdownlib')
+module.cfg(1)
+
+test_run:cmd('switch default')
+test_run:cmd('stop server test')
+os.execute(string.format("grep -r \"stop module fiber\" on_shutdown.log"))
+os.execute(string.format("grep -r \"join module fiber\" on_shutdown.log"))
+os.execute(string.format("grep -r \"module_fiber_f finished\" \
+			 on_shutdown.log"))
+os.execute(string.format("grep -r \"join module fiber finished\" \
+			 on_shutdown.log"))
+test_run:cmd('start server test')
+test_run:cmd('switch test')
+
+box.ctl.set_on_shutdown_timeout(1)
```

Comment:
Nit: in test you use fiber_sleep(1). Just in case I'd make distance between these values a bit greater, so this test will flack under no circumstances.

Reply:
Server test starts two times: at first run we not set box.ctl.set_on_shutdown_timeout, so default timeout == 3sec, and module timeout == 1, here webox.ctl.set_on_shutdown_timeout == 1 and module timeout == 10, specially to check error! I add comments  and direct box.ctl.set_on_shutdown_timeout(3) for first run.

Reply:
Ok, I got it. Then I suggest to do this:
box.ctl.set_on_shutdown_timeout(expected_timeout)
module.cfg(real_timeout)